google-cloud-errors/lib/google/cloud/errors.rb

```diff
@@ -118,6 +118,59 @@ def status_details
         cause.status_details
       end
 
+      ##
+      # Returns the `::Google::Rpc::ErrorInfo` object present in `status_details` array,
+      # given that the following is true:
+      #   * status_details exists and is an array
+      #   * there is only one `::Google::Rpc::ErrorInfo` object among those in the array
+      #
+      # @return [::Google::Rpc::ErrorInfo, nil]
+      def error_info
+        return nil unless status_details.is_a? Array
+        error_infos = status_details.find_all { |status| status.is_a?(::Google::Rpc::ErrorInfo) }
+        return nil unless error_infos.length == 1
+        error_infos[0]
+      end
+
+      ##
+      # Returns the value of `domain` from the `::Google::Rpc::ErrorInfo`
+      # object, if it exists in the `status_details` array.
+      #
+      # This is typically present on errors originating from calls to an API
+      # over gRPC.
+      #
+      # @return [Object, nil]
+      def domain
+        return nill unless error_info.respond_to? :domain
```

Comment:
typo: `nill`

Reply:
I would also add a unit test covering this line.

Reply:
:+1: 
done and done